src/useCapabilitiesSafe.test.ts

```diff
@@ -0,0 +1,108 @@
+import { renderHook } from '@testing-library/react';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import { useCapabilitiesSafe } from './useCapabilitiesSafe';
+
+vi.mock('wagmi', () => ({
+  useAccount: vi.fn(),
+}));
+
+vi.mock('wagmi/experimental', () => ({
+  useCapabilities: vi.fn(),
+}));
+
+describe('useCapabilitiesSafe', () => {
+  const mockChain = {
+    id: 1,
+  };
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
```

Comment:
Open question, what are we missing to test?

Mostly what could go wrong? and we should add some negative tests to check crazy scenarios.